cmd/cue/cmd/testdata/script/issue2969.txtar

```diff
@@ -0,0 +1,37 @@
+# Test for go get issue with time.Duration returning invalid numeric.
+#
+# cue get go
+exec cue get go --local ./...
+cmp ./pkg1/file1_go_gen.cue ./pkg1/file1_go_gen.cue.golden
+
+# Verify dependencies did not change
+cmp go.mod go.mod.golden
+
+-- go.mod --
+module mod.test
+
+go 1.21
+-- go.mod.golden --
+module mod.test
+
+go 1.21
+-- cue.mod/module.cue --
+module: "mod.test"
+-- pkg1/file1.go --
+package pkg1
+
+import "time"
+
+const (
+	Time1Second = time.Duration(time.Second)
+)
+-- pkg1/file1_go_gen.cue.golden --
+// Code generated by cue get go. DO NOT EDIT.
+
+//cue:generate cue get go mod.test/pkg1
+
+package pkg1
+
+import "time"
+
+#Time1Second: time.Duration & "1000000000ns"
```

Comment:
it seems to me like this should be `1s`; perhaps we can have a bit of logic that uses larger units as long as no info is lost. for example, `3000ms` could be `3s`, but `3030ms` could remain as-is. We could also consider e.g. `3m3ms` for the latter case, but that feels like diminishing returns.

Reply:
I think that makes good sense for the humans. In https://github.com/cue-lang/cue/pull/2984/commits/fa8b5594062955ab4b7a28676176cd55dc273e92 I figured we could just parse the string back into a proper `time.Duration`, and use `String()` to pull it back out in a reduced form.

Reply:
Reusing Duration.String is clever :) SGTM